### PR TITLE
Merchant info endpoint groundwork

### PR DESCRIPTION
**Pull request by jurgenwerk**
Ticket: [CS-1432](https://linear.app/cardstack/issue/CS-1432/hub-endpoint-to-create-merchantinfo-generates-and-returns-a-did)

This PR includes all the boilerplate code of adding a new route and a very basic endpoint functionality (insert to DB via POST). 

This is only a part of the bigger feature. I think splitting this feature up into smaller PRs is more convenient to review. 

Missing things (will be added later, probably in a separate PR):
1. Validations
2. DID creation 
3. More comprehensive tests
4. Refactor / DRY up (currently some code is duplicated - serialization, validation stuff)

### CODE REVIEW

File: packages/hub/config/structure.sql
Comment (jurgenwerk) [inline packages/hub/config/structure.sql:600]: In the specification, the name for this field is `unique id`. That sounds a bit confusing to me, because `id` is the actually the unique id. So I took the liberty to use the "slug" nomenclature. Let me know if that's ok!

File: packages/hub/db/migrations/20210809113449561_merchant-infos.js
Comment (jurgenwerk) [inline packages/hub/db/migrations/20210809113449561_merchant-infos.js:14]: Unique index assures there will be no collisions here. 

However, I just noticed this will allow inserting same strings in different case (upper/lower) which will be a problem. I'm thinking about forcing lowercase before persisting. Thoughts?

**Comment by jurgenwerk** [comment]
@lukemelia can you check and see if this PR looks ok - especially the fields and their properties in the migration? In the meanwhile I'll continue working on the DID part.

File: packages/hub/db/migrations/20210809113449561_merchant-infos.js
Comment (lukemelia) [reply packages/hub/db/migrations/20210809113449561_merchant-infos.js:14]: +1 to unique index. I suggest considering using the `citext` module. I believe it is supported on AWS Postgres RDS.

**Review by lukemelia — APPROVED** [review]
On the right track. I left a couple of notes about the database questions. Once addressed, good to merge. 👍

File: packages/hub/db/migrations/20210809113449561_merchant-infos.js
Comment (jurgenwerk) [reply packages/hub/db/migrations/20210809113449561_merchant-infos.js]: Thanks, didn't know this exists! 

I'm a bit reluctant to include new DB extensions unless absolutely necessary (there could be installation issues in various environments, and it's usually something extra to take care of during db provisioning). In this case it can be a simple `toLowercase` when inserting the payload and when filtering. Let me know if you think there's a better solution, but I'm pretty happy with the one I mentioned (will implement this a bit later).

**Comment by jurgenwerk** [comment]
@lukemelia I changed the field names as per your suggestion.

I think this is good to merge now, will do it soon if you agree?

AFAIK the merge will automatically deploy to staging, is that right? Are there any other manual steps included, for example the migration? I guess this is also a good time for me to get familiar with [waypoint](https://linear.app/cardstack/issue/CS-1449/doc-staging-waypoint-setup-last-updated-august-2-2021) so I can learn to do hub deploys 🙃

**Pull request merged**
